domains/client/block-preprocessor/src/inherents.rs

```diff
@@ -12,62 +12,153 @@
 //! Deriving these extrinsics during fraud proof verification should be possible since
 //! verification environment will have access to consensus chain.
 
-use crate::runtime_api::TimestampExtrinsicConstructor;
 use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
-use sp_domains::DomainsApi;
-use sp_inherents::CreateInherentDataProviders;
-use sp_runtime::traits::{Block as BlockT, NumberFor};
+use sp_domains::{DomainId, DomainsApi, DomainsDigestItem};
+use sp_inherents::{CreateInherentDataProviders, InherentData, InherentDataProvider};
+use sp_runtime::traits::{Block as BlockT, Header, NumberFor};
 use sp_timestamp::InherentType;
 use std::error::Error;
-use std::marker::PhantomData;
 use std::sync::Arc;
 
-/// Returns required inherent extrinsics for the domain block based on the primary block.
-/// Note: consensus block hash must be used to construct domain block.
-pub(crate) fn construct_inherent_extrinsics<Block, DomainRuntimeApi, CBlock, CClient>(
-    consensus_client: &Arc<CClient>,
-    domain_runtime_api: &DomainRuntimeApi,
+pub async fn get_inherent_data<CClient, CBlock, Block>(
+    consensus_client: Arc<CClient>,
     consensus_block_hash: CBlock::Hash,
-    domain_parent_hash: Block::Hash,
-) -> Result<Vec<Block::Extrinsic>, sp_blockchain::Error>
+    parent_hash: Block::Hash,
+    domain_id: DomainId,
+) -> Result<InherentData, sp_blockchain::Error>
 where
+    CBlock: BlockT,
     Block: BlockT,
+    CClient: ProvideRuntimeApi<CBlock> + HeaderBackend<CBlock>,
+    CClient::Api: DomainsApi<CBlock, NumberFor<Block>, Block::Hash>,
+{
+    let create_inherent_data_providers =
+        CreateInherentDataProvider::new(consensus_client, Some(consensus_block_hash), domain_id);
+    let inherent_data_providers = <CreateInherentDataProvider<_, _> as CreateInherentDataProviders<
+        Block,
+        (),
+    >>::create_inherent_data_providers(
+        &create_inherent_data_providers, parent_hash, ()
+    )
+        .await?;
```

Comment:
This seems before-`cargo fmt` code, I'm surprised CI doesn't complain about it.

Reply:
`cargo fmt` doesn't complain to me

Reply:
Yeah, since it can pass CI, and that is what I surprised.